### PR TITLE
Fix flac-in-ogg handling

### PR DESCRIPTION
**Pull request by ankith26**
The first commit of this PR is for fixing the detection of FLAC files when it is encapsulated in an OGG transport later (currenty these kind of files go down the Ogg-Vorbis path, which cannot handle it).

The second commit fixes the handling of such files in the libflac backend (the drflac backend seems to already take care of this).

I wrote and tested these commits against the SDL2 branch. With few small porting changes I'm making this PR against `main`. If the maintainers wish I'd be happy to also open the SDL2 branch backport PR for this.

### CODE REVIEW

**Comment by sezero** [comment]
I'm not against this, but not sure how common are the ogg-flac files are.

In any case, shouldn't we handle the extension in `Mix_LoadMUS` too? IIRC,
ogg-flac files usually have `oga` extension.

**Comment by ankith26** [comment]
I'm not sure we can assume every `.oga` file is a flac-in-ogg. There can be even more obscure formats that use ogg as the container format. I think it's instead better to go down the detector path instead of making incorrect guesses

**Comment by sezero** [comment]
OK then. @slouken and/or @icculus: Please review + decide for this one.

**Review by sezero — APPROVED** [review]

**Comment by sezero** [comment]
> OK then. @slouken and/or @icculus: Please review + decide for this one.

@slouken: PING: Do we want this one?

File: src/codecs/music_flac.c [hunk 1]
```diff
@@ -489,6 +511,14 @@ static void *FLAC_CreateFromRW(SDL_RWops *src, SDL_bool freesrc)
...
     int was_error = 1;
     FLAC__int64 full_length;
+    int is_ogg_flac;
+    Uint8 magic[4];
+    if (SDL_RWread(src, magic, 4) != 4) {
+        SDL_SetError("Couldn't read first 4 bytes of audio data");
+        return NULL;
+    }
+    SDL_RWseek(src, -4, SDL_RW_SEEK_CUR);
+    is_ogg_flac = !SDL_memcmp(magic, "OggS", 4);
```
Comment (slouken) [inline src/codecs/music_flac.c:521]: This shouldn't be treated as a boolean, instead please do (SDL_memcmp(magic, "OggS", 4) == 0)

File: src/codecs/music_flac.c [hunk 2]
```diff
@@ -500,17 +530,30 @@ static void *FLAC_CreateFromRW(SDL_RWops *src, SDL_bool freesrc)
...
-                    music) == FLAC__STREAM_DECODER_INIT_STATUS_OK) {
+        if (is_ogg_flac) {
+            ret = flac.FLAC__stream_decoder_init_ogg_stream(
+                music->flac_decoder,
+                flac_read_music_cb, flac_seek_music_cb,
+                flac_tell_music_cb, flac_length_music_cb,
+                flac_eof_music_cb, flac_write_music_cb,
+                flac_metadata_music_cb, flac_error_music_cb,
+                music);
+        }
```
Comment (slouken) [inline src/codecs/music_flac.c:546]: SDL style is else on the same line: "} else {"

**Comment by slouken** [comment]
Conceptually this is great. I added a couple of style notes and we can merge this when they're complete.

**Comment by ankith26** [comment]
I have fixed formatting issues. Also let me know if I can open a backport to SDL2 for this PR

**Comment by sezero** [comment]
This is in, thanks!

>  if I can open a backport to SDL2 for this PR

Please do. (It's only different in two lines..)

**Comment by sezero** [comment]
> > if I can open a backport to SDL2 for this PR
> 
> Please do. (It's only different in two lines..)

Well, I went ahead and cherry-picked it to SDL2 branch as https://github.com/libsdl-org/SDL_mixer/commit/cef595086bf20ca12b31d591fbaaf212721d0b81
Please confirm.

**Comment by ankith26** [comment]
Yup, that looks good to me

**Pull request merged**